### ISSUE TITLE
doesn't emit close event

### ISSUE THREAD

**Issue by klepthys** [gotwarlost/mock-spawn#4]
simple use case, in node : 

    var spawn = require('child_process').spawn;
    child=spawn("cmd");
    child.on("close",function(){
      console.log('close')
    });
    child.on("error",function(){
      console.log("error");
    });

I get output: 

    error
    close

Now, when I do : 
    
    var mock = require("mock-spawn");   
    var spawn = mock();
    spawn.setDefault(mspawn.simple(127));
    /** same code as before**/

"close" event doesn't get called.
My workaround is to specifically call ```emit("close",exit_code);```. Is this normal for some reason to have to do this?

**Comment by gotwarlost** [comment]
Hmmm... it should -https://github.com/gotwarlost/mock-spawn/blob/master/index.js#L20

Not sure why this isn't happening. I made a few changes on Jan 6. Are you using the latest version. If you are, is a previous version doing this right?

**Comment by kerspoon** [comment]
same for me. it works in:
```
mock-spawn@0.2.1 node_modules/mock-spawn
└── through@2.3.6
```

and fails on:
```
mock-spawn@0.2.4 node_modules/mock-spawn
└── through@2.3.6
```

**Comment by gcochard** [comment]
I am seeing the `close` event on node v0.8.28 and *not* on v0.10.38 or v0.12.4 with the following code:

```js
var mockSpawn = require('../../')
  , mySpawn = mockSpawn();

mySpawn.setDefault(function(done) {
  process.nextTick(function(){
    console.log('end!');
    done(0);
  });
});

var proc = mySpawn();

proc.once('close', function(code) {
  console.log('Closed with code', code);
});

proc.once('exit', function(code, signal) {
    console.log(signal?'Ended with signal %s':'Ended with code %s',signal||code);
});

proc.once('error', function(err) {
  console.log('Error with message', err.message);
});
```

Output is as follows:

Node v0.8.28
```
end!
Ended with code 0
Closed with code 0
```

Node v0.10.38
```
end!
Ended with code 0
```

**Comment by xiphiaz** [comment]
:+1: reverting to 0.2.1 fixes the emitting of close event

**Comment by gcochard** [comment]
I think the code in #7 may fix this and #4. I will do some research.
